src/providers/cachedProvider.ts

```diff
@@ -0,0 +1,159 @@
+import { CachingMechanismInterface } from "../interfaces";
+import { RateLimitedProvider } from "./rateLimitedProvider";
+import {
+  BLOCK_NUMBER_TTL,
+  CacheType,
+  lodash,
+  PROVIDER_CACHE_TTL,
+  PROVIDER_CACHE_TTL_MODIFIER as ttl_modifier,
+} from "./utils";
+
+export class CacheProvider extends RateLimitedProvider {
+  public readonly getBlockByNumberPrefix: string;
+  public readonly getLogsCachePrefix: string;
+  public readonly callCachePrefix: string;
+  public readonly baseTTL: number;
+
+  constructor(
+    providerCacheNamespace: string,
+    readonly redisClient?: CachingMechanismInterface,
```

Comment:
Note: we didn't have a `RedisClient` instance in the SDK but we do have `CachingMechanismInterface` which should be able to take a `RedisClient`

Reply:
whats the idea behind this interface?

Reply:
also assuming this is mainly copied from the other repo?

Reply:
THe `CachingMechanismInterface` is a way the sdk can stay un-opinionated about the actual persistence layer while still serving basic get/set functionality. Think: Redis as a cache or Arweave as a cache (which we do in the relayer)

Reply:
Yes, all 3 classes have been copied - the corresponding relayer-v2 PR removes the logic